gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnAppLauncher.java

```diff
@@ -380,6 +382,22 @@ public void launch() throws IOException, YarnException, InterruptedException {
     }, 0, this.appReportIntervalMinutes, TimeUnit.MINUTES);
 
     addServices();
+
+    // The YarnClient and all the services are started asynchronously.
+    // This will block until the application is completed and throws an exception to fail the Azkaban Job in case the
+    // underlying Yarn Application reports a job failure.
+    synchronized (this.applicationDone) {
+      while (!this.applicationCompleted) {
+        try {
+          this.applicationDone.wait();
```

Comment:
it might be simpler and cleaner to use `CountDownLatch` instead of explicit synchronization with `synchronized, wait(), and notify()`